style: improve contact editor design

SuhasDissa opened the pull request:
| Before | After |
| --- | --- |
| ![Screenshot_20240103_132844](https://github.com/you-apps/ConnectYou/assets/64766434/620e7170-5c8d-413c-a5cb-022235ee9db2) | ![Screenshot_20240103_164439](https://github.com/you-apps/ConnectYou/assets/64766434/d51675a1-1cd0-43dd-b696-c005f9a36cef) |

Bnyro commented on app/src/main/java/com/bnyro/contacts/ui/components/editor/TextFieldGroup.kt:
```diff
@@ -0,0 +1,81 @@
+import com.bnyro.contacts.obj.ValueWithType
+
+@Composable
+fun SnapshotStateList<MutableState<ValueWithType>>.TextFieldGroup(
```
```suggestion
fun TextFieldGroup(
```

Bnyro replied:
I would prefer passing this as a normal argument, I don't think it's common to use extension functions for composables like that.

Bnyro commented on app/src/main/java/com/bnyro/contacts/ui/components/editor/TextFieldEditor.kt:
```diff
@@ -21,28 +39,72 @@ fun TextFieldEditor(
+                Icon(
+                    imageVector = Icons.Rounded.Remove, contentDescription = stringResource(
+                        id = R.string.delete
+                    ), tint = Color.Red
```
```suggestion
                    ), tint = MaterialTheme.colorScheme.error
```

Bnyro commented:
You probably forgot to re-add the account type dialog?

Bnyro commented:
I really like the visual changes btw.

My only concern is thr color of the text fields, they loom a bit to gray-ish to me personally.

SuhasDissa commented:
> You probably forgot to re-add the account type dialog?

I removed the dialog and added a drop-down menu inside the top app bar. It's not visible if there's only one account present.
IMO, It's better to have the account switch in a more prominent place

SuhasDissa commented:
> My only concern is thr color of the text fields, they loom a bit to gray-ish to me personally.

It's the default color of material 3 text field. Changing that is quite frustrating

Bnyro commented on app/src/main/java/com/bnyro/contacts/ui/components/editor/TextFieldGroup.kt:
```diff
@@ -0,0 +1,86 @@
+                    shape = if (index == 0) RoundedCornerShape(50, 50, 0, 0) else RectangleShape
+                )
+            }
+            AnimatedVisibility(true) {
```
What's that line used for?

SuhasDissa replied:
Some mistake :)
